### ISSUE TITLE
Enable better IntelliSense in STM32Cube framework by adding Src path to search path

### ISSUE THREAD

**Issue by maxgerhardt** [platformio/platform-ststm32#449]
As explained in https://community.platformio.org/t/a-detailed-description-of-the-functions-in-the-platformio/17369/5. 

When you create a STM32Cube project in VSCode, e.g. for a board of the L4 series, you don't get the full intended autocomplete, that is, function descriptions, parameter descriptions etc are missing. This does work perfectly in fine in the competitor CubeIDE.

The source of the problem is that the STM32HAL has the actually detailed Doyxgen documentation in the `Src/` folder of the HAL, not in the `Inc/` folder, which is currently that what PlatformIO includes in the search path. As can be seen in the topic, after manually adding `../l4/Drivers/STM32L4xx_HAL_Driver/Src` in the `c_cpp_properties.json`, the IntelliSense goes from

![grafik](https://user-images.githubusercontent.com/26485477/100235968-56217880-2f2d-11eb-978a-f735b98514b3.png)

to

![grafik](https://user-images.githubusercontent.com/26485477/100235975-591c6900-2f2d-11eb-9680-b7d3e01e0753.png)

This should be easily fixable by adding the `Src/` path too in the `stm32cube.py` builder script.

**Comment by maxgerhardt** [comment]
Oh wow that was a **massive** update, also all the STM32Cube packets were split up now according to their STM series? The underlying STM32HAL package was updated as well, thus resolving https://github.com/platformio/platform-ststm32/issues/439?

**Comment by valeros** [comment]
You're right, we now use the official packages. I also added some tweaks to configure the framework in a bit more flexible manner.
